runs/models.tsp

```diff
@@ -483,4 +501,163 @@ model RunToolCallObject {
+  RunStepDeltaStepDetailsToolCallsCodeOutputLogsObject,
+  RunStepDeltaStepDetailsToolCallsCodeOutputImageObject,
+}
+
+/** Details of a code interpreter tool call that a streaming run step was involved in. */
+model RunStepDeltaStepDetailsToolCallsCodeObject {
+  /** The index of the tool call in the tool calls array. */
+  index: safeint;
+
+  /** The ID of the tool call. */
+  id: string;
+
+  /** The type of the tool call, which is always `code_interpreter`.
+  type: "code_interpreter";
+
+  /** The Code Interpreter tool call definition. */
+  code_interpreter: {
+    /** The input to the Code Interpreter tool call. */
+    input: string;
+
+    /**
+     * The outputs from the Code Interpreter tool call. Code Interpreter can output one or more
+     * items, including text (`logs`) or images (`image`). Each of these are represented by a
+     * different object type.
+     */
+    outputs?: RunStepDeltaStepDetailsToolCallsCodeOutput[];
+  }
+}
+
+/**
+ * Details of log outputs from a Code Interpreter tool call within a streaming run step.
+ */
+model RunStepDeltaStepDetailsToolCallsCodeOutputLogsObject {
+  /** The index of the output in the outputs array. */
+  index: safeint;
+
+  /** The type of the object, which is always `logs`. */
+  type: "logs";
+
+  /** The text output from the Code Interpreter tool call. */
+  logs?: string;
+}
+
+/**
+ * Details of an image output from a Code Interpreter tool call within a streaming run step.
+ */
+model RunStepDeltaStepDetailsToolCallsCodeOutputImageObject {
+  /** The index of the output in the outputs array. */
+  index: safeint;
+
+  /** The type of the object, which is always `image`. */
+  type: "image";
+
+  /** The file ID data for the image. */
+  image?: {
+    /** The file ID for the image. */
+    file_id?: string;
+  };
+}
+
+/** Details of a retrieval tool call that a streaming run step was involved in. */
+model RunStepDeltaStepDetailsToolCallsRetrievalObject {
+  /** The index of the tool call in the tool calls array. */
+  index: safeint;
+
+  /** The ID of the tool call. */
+  id: string;
+
+  /** The type of the tool call, which is always `retrieval`. */
+  type: "retrieval";
+
+  /** Reserved for future use. */
+  @extension("x-oaiTypeLabel", "map")
+  retrieval?: Record<string> | null;
+}
+
+/** Details of a function tool call that a streaming run step was involved in. */
+model RunStepDeltaStepDetailsToolCallsFunctionObject {
+  /** The index of the tool call in the tool calls array. */
+  index: safeint;
+
+  /** The ID of the tool call. */
+  id: string;
+
+  /** The type of the tool call, which is always `retrieval`. */
+  type: "function";
+
+  /** The function data for the tool call. */
+  function: {
+    /** The name of the function. */
+    name: string;
+
+    /** The arguments passed to the function. */
+    arguments: string;
+
+    /**
+     * The output of the function. This will be `null` if the outputs have not yet been submitted.
```

Comment:
nit:

```text
The output of the function. This will be `null` if the outputs have not been [submitted](/docs/api-reference/runs/submitToolOutputs) yet.
```

Reply:
Do we handle markdown properly in that manner? I thought embedding docs was one of our limitations.

Reply:
Ah, I see we already have them all over the place anyhow -- they indeed do not appear to work well, but given this is (a) the "internal" part and (b) consistent with the rest, I'll add it in. We'll want to figure out how to deal with this when we move to direct use of the generated models, though.